Various plugin unavailability codepaths in HTMLPlugInElement/RenderEembeddedObject are unused

aprotyas opened the pull request:
#### e99fbe347168f2738d52ea51626ea37f27bb284e
<pre>
Various plugin unavailability codepaths in HTMLPlugInElement/RenderEembeddedObject are unused
<a href="https://bugs.webkit.org/show_bug.cgi?id=286567">https://bugs.webkit.org/show_bug.cgi?id=286567</a>
<a href="https://rdar.apple.com/143686152">rdar://143686152</a>

Reviewed by Anne van Kesteren and Tim Horton.

Following 243690@main and 282972@main, quite a few methods (and class
states) corresponding to plugin unavailability are unused. This PR
removes all this dead code.

* Source/WebCore/html/HTMLPlugInElement.cpp:
(WebCore::HTMLPlugInElement::setReplacement): Deleted.
(WebCore::HTMLPlugInElement::isReplacementObscured): Deleted.
* Source/WebCore/html/HTMLPlugInElement.h:
* Source/WebCore/rendering/RenderEmbeddedObject.cpp:
(WebCore::RenderEmbeddedObject::setPluginUnavailabilityReason):
(WebCore::RenderEmbeddedObject::paintReplaced):
(WebCore::RenderEmbeddedObject::getCursor const):
(WebCore::RenderEmbeddedObject::setPluginUnavailabilityReasonWithDescription): Deleted.
(WebCore::RenderEmbeddedObject::setUnavailablePluginIndicatorIsHidden): Deleted.
(WebCore::RenderEmbeddedObject::unavailablePluginIndicatorBounds const): Deleted.
* Source/WebCore/rendering/RenderEmbeddedObject.h:
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::isPluginUnavailabilityIndicatorObscured): Deleted.
(WebCore::Internals::unavailablePluginReplacementText): Deleted.
(WebCore::Internals::isPluginSnapshotted): Deleted.
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:

Canonical link: <a href="https://commits.webkit.org/289485@main">https://commits.webkit.org/289485@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0accbec774ccf0c0f43e8e2f67c1ed5346033ed5

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/87073 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/6583 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/41431 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/91932 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/37812 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/89123 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/6859 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/14651 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/67300 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/25054 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/90075 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/5254 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/78819 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/47622 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/5030 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/33195 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/36929 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/75521 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/34074 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/93819 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/14235 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/10361 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/76103 "Found 2 new test failures: imported/w3c/web-platform-tests/css/css-view-transitions/active-view-transition-type-on-non-root.html imported/w3c/web-platform-tests/service-workers/service-worker/redirected-response.https.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/14439 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/74675 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/75303 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/19644 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/18081 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/7152 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/13567 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/14254 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/19547 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/13999 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/17441 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/15780 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebCore/html/HTMLPlugInElement.cpp:
```diff
@@ -374,26 +374,6 @@ bool HTMLPlugInElement::requestObject(const String& relativeURL, const String& m
-    bool replacementIsObscured = isReplacementObscured();
-    // hittest in isReplacementObscured() method could destroy the renderer. Let's refetch it.
-    if (CheckedPtr renderer = dynamicDowncast<RenderEmbeddedObject>(this->renderer()))
-        renderer->setUnavailablePluginIndicatorIsHidden(replacementIsObscured);
```
This is the last call for this method.

annevk commented on Source/WebCore/html/HTMLPlugInElement.cpp:
```diff
@@ -374,26 +374,6 @@ bool HTMLPlugInElement::requestObject(const String& relativeURL, const String& m
-
-        renderer->setPluginUnavailabilityReasonWithDescription(reason, unavailabilityDescription);
-    }
-    bool replacementIsObscured = isReplacementObscured();
```
This seems to be the last non-test call for this method.

aprotyas replied:
Indeed. Also, the Internals test API is not invoked by any test, so we can also remove that.

annevk commented on Source/WebCore/html/HTMLPlugInElement.cpp:
```diff
@@ -374,26 +374,6 @@ bool HTMLPlugInElement::requestObject(const String& relativeURL, const String& m
-        if (reason == PluginUnavailabilityReason::UnsupportedPlugin)
-            document().addConsoleMessage(MessageSource::JS, MessageLevel::Warning, "Tried to use an unsupported plug-in."_s);
-
-        renderer->setPluginUnavailabilityReasonWithDescription(reason, unavailabilityDescription);
```
This seems to be the last call for this method.

aprotyas replied:
This method is still called from `RenderEmbeddedObject::setPluginUnavailabilityReason()`, which is used by the `SubframeLoader` class.

annevk reviewed (approved):
I like this, but more cleanup seems possible.

annevk commented on Source/WebCore/html/HTMLPlugInElement.cpp:
```diff
@@ -374,97 +374,6 @@ bool HTMLPlugInElement::requestObject(const String& relativeURL, const String& m
-    }
-    // Calculate the absolute rect for the blocked plugin replacement text.
-    LayoutPoint absoluteLocation(pluginRenderer->absoluteBoundingBoxRect().location());
-    LayoutRect rect = pluginRenderer->unavailablePluginIndicatorBounds(absoluteLocation);
```
I think this can be removed as well now.

annevk reviewed (approved):
Very nice, I think I spotted one more.

aprotyas committed:
Various plugin unavailability codepaths in HTMLPlugInElement/RenderEembeddedObject are unused
https://bugs.webkit.org/show_bug.cgi?id=286567
rdar://143686152

Reviewed by Anne van Kesteren and Tim Horton.

Following 243690@main and 282972@main, quite a few methods (and class
states) corresponding to plugin unavailability are unused. This PR
removes all this dead code.

* Source/WebCore/html/HTMLPlugInElement.cpp:
(WebCore::HTMLPlugInElement::setReplacement): Deleted.
(WebCore::HTMLPlugInElement::isReplacementObscured): Deleted.
* Source/WebCore/html/HTMLPlugInElement.h:
* Source/WebCore/rendering/RenderEmbeddedObject.cpp:
(WebCore::RenderEmbeddedObject::setPluginUnavailabilityReason):
(WebCore::RenderEmbeddedObject::paintReplaced):
(WebCore::RenderEmbeddedObject::getCursor const):
(WebCore::RenderEmbeddedObject::setPluginUnavailabilityReasonWithDescription): Deleted.
(WebCore::RenderEmbeddedObject::setUnavailablePluginIndicatorIsHidden): Deleted.
(WebCore::RenderEmbeddedObject::unavailablePluginIndicatorBounds const): Deleted.
* Source/WebCore/rendering/RenderEmbeddedObject.h:
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::isPluginUnavailabilityIndicatorObscured): Deleted.
(WebCore::Internals::unavailablePluginReplacementText): Deleted.
(WebCore::Internals::isPluginSnapshotted): Deleted.
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:

Canonical link: https://commits.webkit.org/289485@main
